torch/package/package_exporter.py

```diff
@@ -493,6 +499,29 @@ def save_module(self, module_name: str, dependencies=True):
 
         self._intern_module(module_name, dependencies)
 
+    def _is_package(
+        self,
+        module_obj: types.ModuleType
+    ):
+        """Checks if module_obj is a package and throws an error, if module_obj is an
+            invalid package.
+        """
+        error = BadPackageError(f"{module_obj.__name__} is an improperly formed package "
+                                f",so it cannot be interned. Please mock or extern {module_obj.__name__}.")
+        if not hasattr(module_obj, "__path__"):
+            return False
+        elif isinstance(module_obj.__path__, list):  # type: ignore [attr-defined]
+            return True
+        try:
+            for i in module_obj.__name__:
```

Comment:
I don't understand what we're testing here, I assume it has something to do with iterableness; can you add a comment explaining